onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -220,59 +220,11 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<onnxruntime:
     if (!func_body) {
       return common::Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT, "Function body is empty");
     }
-    const Graph& graph_body = func_body->Body();
-    onnxruntime::Model model(graph_body.Name(), true, ModelMetaData(), IOnnxRuntimeOpSchemaRegistryList(), graph_body.DomainToVersionMap());
-    onnxruntime::Graph& graph = model.MainGraph();
-
-    for (const auto& graph_body_node : graph_body.Nodes()) {
-      graph.AddNode(graph_body_node);
-    }
-
-    ORT_ENFORCE(graph.Resolve().IsOK());
-
-    // Add initializer to graph
-    const auto& init_tensors = graph_body.GetAllInitializedTensors();
-    for (const auto& tensor : init_tensors) {
-      graph.AddInitializedTensor(*(tensor.second));
-    }
+    auto& graph_body = func_body->Body().ToGraphProto();
```

Comment:
The func_body->Body() returns const object while ToGraphProto() requires to change it. That's why we reconstruct the graph.

Reply:
I added an interface for this purpose :).

Reply:
see, function.h change

Reply:
There is still error,
/onnxruntime_src/onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:223:55: error: passing 'const onnxruntime::Graph' as 'this' argument discards qualifiers [-fpermissive]
     auto& graph_body = func_body->Body().ToGraphProto();

Reply:
probably because fused_node->GetFunctionBody also returns constant object func_body

Reply:
Thanks for the comments. yes. the APIs need to be changed a little bit.